go_covert_lib/client/src/App.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import Button from 'react-bootstrap/Button';
+import FormControl from 'react-bootstrap/FormControl';
+import InputGroup from 'react-bootstrap/InputGroup';
+import './styles.css';
+
+/**
+ * IMPORTANT NOTE: For styling, refer to https://getbootstrap.com/docs/4.0/utilities/position/
+ */
+
+const App = () => {
+  const [textToSend, setTextToSend] = useState('');
+  const [channels, setChannels] = useState([]);
+  let ws; // The websocket connection to the server
+
+  const sendConfig = () => {
+    const cmd = JSON.stringify({ OpCode: 'config' });
+    ws.send(cmd);
+  };
+
+  useEffect(() => {
+    // Matches just the "127.0.0.1:8080" portion of the address
+    const addressRegex = /[a-zA-Z0-9.]+:[\d]+/g;
+    ws = new WebSocket(`ws://${window.location.href.match(addressRegex)[0]}/api/ws`);
+    ws.binaryType = 'arraybuffer';
+    ws.onopen = (event) => {
+      sendConfig();
+    };
+
+    ws.onerror = (event) => {
+      // TODO:
+    };
+
+    ws.onmessage = (event) => {
+      const msg = JSON.parse(event.data);
+      switch (msg.OpCode) {
+        case 'config':
+          setChannels(Object.keys(msg.Default.Channel));
+          break;
+        default:
+            // TODO:
+      }
+    };
+  }, []);
+
+  return (
```

Comment:
Everything below here is placeholder templating which will instead of being hardcoded in, will be dynamically generated based on parsing the JSON.